packages/salesforcedx-utils-vscode/test/unit/helpers/traceFlags.test.ts

```diff
@@ -155,6 +155,84 @@ describe('Trace Flags', () => {
     expect(expDate.getTime() - currDate).to.be.greaterThan(60000 * 29);
   });
 
+  it('should verify that createTraceFlag() is called when a trace flag doesn\'t exist', async () => {
+    queryStub = sb.stub(mockConnection, 'query');
+    toolingCreateStub = sb.stub(mockConnection.tooling, 'create');
+    toolingQueryStub = sb.stub(mockConnection.tooling, 'query');
+    toolingUpdateStub = sb.stub(mockConnection.tooling, 'update');
+
+    const traceFlags = new TraceFlags(mockConnection);
+    const createTraceFlagSpy = sb.spy(TraceFlags.prototype, 'createTraceFlag');
+
+    queryStub
+      .onFirstCall()
+      .resolves({
+        done: true,
+        totalSize: 1,
+        records: [{ Id: USER_ID }]
+      });
+    toolingQueryStub
+      .onFirstCall()
+      .resolves({
+        done: true,
+        totalSize: 0,
+        records: []
+      });
+    toolingCreateStub
+      .onFirstCall()
+      .resolves({ success: true, id: 'aBcDeF' })
+      .onSecondCall()
+      .resolves({ success: true, id: '01020304' });
+
+    await traceFlags.ensureTraceFlags();
+    expect(createTraceFlagSpy.callCount).to.equal(1);
+  });
+
+  it('should verify that createTraceFlag() is not called when a trace flag exist', async () => {
```

Comment:
Can we also add a test that we don't delete existing flags, and that we are able to delete the flags that we create?

Reply:
I'm working on tests for the other files I touched.  After I finished I'll circle back to this.  I think the ask of "add a test that we don't delete existing flags" might be covered in another file (either traceFlagsRemover, or this one, traceFlags)

Reply:
I think these two tests cover both of these scenarios.  Let's sync up & discuss.